include/fastdds/rtps/messages/RTPSMessageGroup.h

```diff
@@ -44,195 +44,212 @@ class RTPSMessageGroup_t;
+     * @return True when message was added to the group.
+     */
+    bool add_acknack(
+            const SequenceNumberSet_t& seq_num_set,
+            int32_t count,
+            bool final_flag);
+
+    /**
+     * Adds a NACKFRAG message to the group.
+     * @param seq_number Sequence number being nack'ed.
+     * @param fn_state Set of missing fragment numbers.
+     * @param count Counting identifier.
+     * @return True when message was added to the group.
+     */
+    bool add_nackfrag(
+            const SequenceNumber_t& seq_number,
+            FragmentNumberSet_t fn_state,
+            int32_t count);
+
+    inline uint32_t get_current_bytes_processed() const
+    { 
+        return currentBytesSent_ + full_msg_->length; 
+    }
+
+    /**
+     * To be used whenever destination locators/guids change between two add_xxx calls.
+     * Automatically called inside add_xxx calls if destinations_have_changed() method of
+     * RTPSMessageSenderInterface returns true.
+     * May become private again with a refactor of RTPSMessageSenderInterface, adding a
+     * group_has_been_flushed() method.
+     */
+    void flush_and_reset();
+
+    //! Maximum fragment size minus the headers
+    static inline constexpr uint32_t get_max_fragment_payload_size()
+    {
+        // Max fragment is 64KBytes_max - header - inlineqos - 3(for better alignment)
+        return std::numeric_limits<uint16_t>::max() - data_frag_header_size_ - max_inline_qos_size_ - 3;
+    }
+
+private:
+
+    static constexpr uint32_t data_frag_header_size_ = 28;
+    static constexpr uint32_t max_inline_qos_size_ = 32;
+
+    void reset_to_header();
+
+    void flush();
+
+    void send();
+
+    void check_and_maybe_flush()
+    {
+        check_and_maybe_flush(sender_.destination_guid_prefix());
+    }
+
+    void check_and_maybe_flush(
+            const GuidPrefix_t& destination_guid_prefix);
+
+    bool insert_submessage(
+            bool is_big_submessage)
+    {
+        return insert_submessage(sender_.destination_guid_prefix(), is_big_submessage);
+    }
+
+    bool insert_submessage(
+            const GuidPrefix_t& destination_guid_prefix,
+            bool is_big_submessage);
+
+    bool add_info_dst_in_buffer(
+            CDRMessage_t* buffer,
+            const GuidPrefix_t& destination_guid_prefix);
+
+    bool add_info_ts_in_buffer(
+            const Time_t& timestamp);
+
+    bool create_gap_submessage(
+            const SequenceNumber_t& gap_initial_sequence,
+            const SequenceNumberSet_t& gap_bitmap,
+            const EntityId_t& reader_id);
+
+    const RTPSMessageSenderInterface& sender_;
+
+    Endpoint* endpoint_;
+
+    CDRMessage_t* full_msg_;
+
+    CDRMessage_t* submessage_msg_;
+
+    uint32_t currentBytesSent_;
+
+    GuidPrefix_t current_dst_;
 
 #if HAVE_SECURITY
-        CDRMessage_t* encrypt_msg_;
+    RTPSParticipantImpl* participant_;
+
+    CDRMessage_t* encrypt_msg_;
```

Comment:
`participant_` is also used without security. This does not build with security off.